File: javascript/raindrops/raindrops.js
'use strict';

export const convert = (number) => {
    const factorToSound = new Map([
        [3, 'Pling'],
        [5, 'Plang'],
        [7, 'Plong']
    ]);

    let drops = "";
    for (const [factor, sound] of factorToSound)
        if (number % factor === 0)
             drops += sound;

    if (drops === "")
        drops += number;

    return drops;
};

File: clojure/raindrops/src/raindrops.clj
(ns raindrops)

(defn convert [number]
  (let [drops (apply str
                     (map (fn [factor string]
                            (if (== (mod number factor) 0)
                              string nil))
                          [3 5 7]
                          ["Pling" "Plang" "Plong"]))]
    (if (= drops "")
      (str number) drops)))
